src/utils/utils.js

```diff
@@ -481,11 +481,33 @@ export const filterDuplicateAgencies = (userInfoAccounts) => {
 /**
  *
  * @param {*} context context
- * @returns an array of users agency ids
+ * @returns an array of users branch ids
  */
 export const getUserBranchIds = async (context) => {
+
+  //get users agencies
   const userinfo = await context.datasources.getLoader("userinfo").load({
     accessToken: context.accessToken,
   });
-  return userinfo?.attributes?.agencies?.map((agency) => agency?.agencyId);
+  const agencies = filterDuplicateAgencies(userinfo?.attributes?.agencies)?.map(
+    (account) => account.agencyId
+  );
+  //fetch branches for each agency
+  const agencyInfos = await Promise.all(
+    agencies.map(
+      async (agency) =>
+        await context.datasources.getLoader("library").load({
+          agencyid: agency,
+          limit: 20,
+          status: "ALLE",
+        })
+    )
+  );
+
+  //merge all branchIds into one array
+  let branchIdsArray = [];
```

Comment:
Tror hvis du bygger din Promise.all med en return kan du kører agencyInfos.flat() for at samle det. 

Se fx orders.datasource

Reply:
flat() er the trick